diff --git a/loginrec.c b/loginrec.c
index 9a427de..f64d43a 100644
--- a/loginrec.c
+++ b/loginrec.c
@@ -436,7 +436,7 @@ login_set_addr(struct logininfo *li, const struct sockaddr *sa,
 int
 login_write(struct logininfo *li)
 {
-#ifndef HAVE_CYGWIN
+#if !defined(HAVE_CYGWIN) && !defined(__PASE__)
 	if (geteuid() != 0) {
 		logit("Attempt to write login records by non-root user (aborting)");
 		return (1);
