diff --git a/platform-misc.c b/platform-misc.c
index 3f39670..b61425b 100644
--- a/platform-misc.c
+++ b/platform-misc.c
@@ -25,6 +25,23 @@
 int
 platform_sys_dir_uid(uid_t uid)
 {
+#ifdef __PASE__
+	static uid_t qsys_uid = -1;
+
+	if(qsys_uid == (uid_t)-1)
+	{
+		struct passwd* pw;
+		struct passwd pw_struct;
+		char pw_buf[1024];
+		getpwnam_r("qsys", &pw_struct, pw_buf, sizeof(pw_buf), &pw);
+		if(pw)
+			qsys_uid = pw->pw_uid;
+		debug3("%s: uid=%d, platform sys uid=%d", __func__, uid, qsys_uid);
+	}
+
+	if (uid == qsys_uid)
+			return 1;
+#endif
 	if (uid == 0)
 		return 1;
 #ifdef PLATFORM_SYS_DIR_UID
