diff --git a/servconf.c b/servconf.c
index 932d363..8c902a8 100644
--- a/servconf.c
+++ b/servconf.c
@@ -180,6 +180,9 @@ initialize_server_options(ServerOptions *options)
 	options->fingerprint_hash = -1;
 	options->disable_forwarding = -1;
 	options->expose_userauth_info = -1;
+#ifdef __PASE__
+	options->pase_shell = NULL;
+#endif
 }
 
 /* Returns 1 if a string option is unset or set to "none" or 0 otherwise. */
@@ -411,6 +414,10 @@ fill_default_server_options(ServerOptions *options)
 		options->disable_forwarding = 0;
 	if (options->expose_userauth_info == -1)
 		options->expose_userauth_info = 0;
+#ifdef __PASE__
+	if (options->pase_shell == NULL)
+		options->pase_shell = xstrdup(_PATH_BSHELL);
+#endif
 
 	assemble_algorithms(options);
 
@@ -497,6 +504,9 @@ typedef enum {
 	sStreamLocalBindMask, sStreamLocalBindUnlink,
 	sAllowStreamLocalForwarding, sFingerprintHash, sDisableForwarding,
 	sExposeAuthInfo, sRDomain,
+#ifdef __PASE__
+	sIBMPASEforiShell, sIBMPASEforiEnv, sIBMPASEforiLangid, sIBMPASEforiCntryid,
+#endif
 	sDeprecated, sIgnore, sUnsupported
 } ServerOpCodes;
 
@@ -645,6 +655,12 @@ static struct {
 	{ "exposeauthinfo", sExposeAuthInfo, SSHCFG_ALL },
 	{ "rdomain", sRDomain, SSHCFG_ALL },
 	{ "casignaturealgorithms", sCASignatureAlgorithms, SSHCFG_ALL },
+#ifdef __PASE__
+	{ "ibmpaseforishell",   sIBMPASEforiShell,   SSHCFG_GLOBAL},
+	{ "ibmpaseforienv",     sIBMPASEforiEnv,     SSHCFG_GLOBAL},
+	{ "ibmpaseforilangid",  sIBMPASEforiLangid,  SSHCFG_GLOBAL},
+	{ "ibmpaseforicntryid", sIBMPASEforiCntryid, SSHCFG_GLOBAL},
+#endif
 	{ NULL, sBadOption, 0 }
 };
 
@@ -1960,6 +1976,18 @@ process_server_config_line(ServerOptions *options, char *line,
 			options->adm_forced_command = xstrdup(cp + len);
 		return 0;
 
+#ifdef __PASE__
+	case sIBMPASEforiShell:
+		charptr = &options->pase_shell;
+		goto parse_filename;
+
+	/* These are ignored */
+	case sIBMPASEforiEnv:
+	case sIBMPASEforiLangid:
+	case sIBMPASEforiCntryid:
+		return 0;
+#endif
+
 	case sChrootDirectory:
 		charptr = &options->chroot_directory;
 
